src/crypto/CHIPCryptoPALOpenSSL.cpp

```diff
@@ -1773,19 +1773,20 @@ CHIP_ERROR ExtractAKIDFromX509Cert(const ByteSpan & certificate, MutableByteSpan
     return ExtractKIDFromX509Cert(false, certificate, akid);
 }
 
-CHIP_ERROR ExtractVIDFromX509Cert(const ByteSpan & certificate, VendorId & vid)
+namespace {
+
+CHIP_ERROR ExtractDNAttributeFromX509Cert(bool isVID, const ByteSpan & certificate, uint16_t & id)
```

Comment:
Suggest an API that has an enum instead of `isVID`, and uint64_t instead of uint16_t for ID. This would scale to any Matter IDs in X509 cert, and would allow:

`ExtractDNAttributeFromX509Cert(MatterOid::kVendorId, cert, dacVendorId)`.

This would also allow extracting future DNs more easily.

Reply:
Changed the input param to an enum.